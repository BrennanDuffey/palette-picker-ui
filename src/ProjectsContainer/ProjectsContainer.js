import React, { Component } from 'react';

class ProjectsContainer extends Component {
  

  render() {
    return(
      <section className="ProjectsContainer">
        <h2>Your Projects</h2>
      </section>
    );
  }
}

export default ProjectsContainer;
