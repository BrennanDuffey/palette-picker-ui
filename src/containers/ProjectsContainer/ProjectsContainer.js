import React, { Component } from 'react';
import ProjectCard from '../../components/ProjectCard/ProjectCard';

class ProjectsContainer extends Component {
  displayProjects = () => {
    let projects;

    if (this.props.projects) {
      projects = projects.map(project => {
        return <p>{project.name}</p>
      });
    } else {
      projects = 
        <article className="empty-projects-card">
          <h3>There are no saved projects!</h3>
        </article>
    }

    return projects;
  }

  render() {
    return (
      <section className="ProjectsContainer">
        <h2>Your Projects</h2>
        {this.displayProjects()}
        <ProjectCard />
      </section>
    );
  }
}

export default ProjectsContainer;
