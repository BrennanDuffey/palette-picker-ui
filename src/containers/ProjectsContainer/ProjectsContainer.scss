.ProjectsContainer {
  background: $lightGray;
  height: 100vh;
  padding: 50px;

  .empty-projects-card {
    background: white;
    font-style: italic;
    font-weight: 300;
    margin: 20px 0;
    padding: 100px;
    text-align: center;
  }

  .add-project-btn {
    background: white;
    border: none;
    border-radius: 5px;
    display: flex;
    flex-direction: row;
    font-size: 1.1em;
    margin: 30px 0;
    padding: 15px;

    p {
      padding: 0 10px;
    }
  }
}