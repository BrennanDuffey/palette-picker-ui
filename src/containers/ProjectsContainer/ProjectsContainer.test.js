import React from 'react';
import { shallow } from 'enzyme';
import ProjectsContainer from './ProjectsContainer';

describe('ProjectsContainer', () => {
  let mockProjects;
  // let mockPalettes;
  let wrapper;

  beforeEach(()=> {
    mockProjects = [
      { 
        id: 1,
        name: 'Project 1', 
      }
    ];
    // mockPalettes = [
    //   {
    //     id: 11,
    //     name: 'Palette 1',
    //     colors_array: [
    //       '#FDC58D', 
    //       '#FFFFAA', 
    //       '#95E38A', 
    //       '#779DF2', 
    //       '#D2BEFD'
    //     ],
    //     project_id: 1
    //   }
    // ];
    wrapper = shallow(
      <ProjectsContainer projects={mockProjects} />
    );
  });

  it('should match snapshot if there are saved projects', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if there are no saved projects', () => {
    wrapper = shallow(
      <ProjectsContainer projects={[]} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});