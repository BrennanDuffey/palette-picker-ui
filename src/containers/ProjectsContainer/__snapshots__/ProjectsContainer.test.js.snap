// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsContainer should match snapshot 1`] = `
<section
  className="ProjectsContainer"
>
  <h2>
    Your Projects
  </h2>
  <article
    className="empty-projects-card"
  >
    <h3>
      You have not created any projects!
    </h3>
  </article>
</section>
`;
